### PR TITLE
Added log warning for when total size of PL arrays for a VC will likely exceed 100,000

### PR DESCRIPTION
**Pull request by KevinCLydon**

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypingEngine.java
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypingEngine.java]: It would be good to include the following things in this message: 
1.  variant position
2. the number of alleles at the site
3. the ploidy
4. the predicted number of pls

**Comment by droazen** [comment]
@KevinCLydon Let's wait until https://github.com/samtools/htsjdk/issues/1446 is addressed before merging this one.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypingEngine.java
Comment (KevinCLydon) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypingEngine.java]: Good point.  Added that extra info

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypingEngine.java
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypingEngine.java]: @KevinCLydon I think we're going to want to change this warning to be a bit more conservative, otherwise for large callsets such as GnomAD we'll get warnings all over the place. We should probably just check the size of the PL array for one sample, and warn if that's too big, rather than multiplying that value by the number of samples.

Comment (KevinCLydon) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypingEngine.java]: That sounds reasonable. Just pushed a new branch with this change

**Review by droazen — APPROVED** [review]
:+1: Approved to merge after switching to a `OneShotLogger`

**Pull request merged**
